// Copyright (c) 2016-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the BSD-style license found in the
// LICENSE file in the root directory of this source tree. An additional grant
// of patent rights can be found in the PATENTS file in the same directory.

syntax = "proto3";

package magma.lte;

import "lte/protos/subscriberdb.proto";
import "lte/protos/policydb.proto";

option go_package = "magma/lte/cloud/go/protos";

message CreateBearerRequest {
    SubscriberID sid = 1;
    string ip_addr = 2;
    uint32 link_bearer_id = 3;
    repeated PolicyRule policy_rules = 4;
}

message CreateBearerResult {}

message DeleteBearerRequest {
    SubscriberID sid = 1;
    string ip_addr = 2;
    uint32 link_bearer_id = 3;
    repeated uint32 eps_bearer_ids = 4;
}

message DeleteBearerResult {}

service SpgwService {
    rpc CreateBearer (CreateBearerRequest) returns (CreateBearerResult) {}
    rpc DeleteBearer (DeleteBearerRequest) returns (DeleteBearerResult) {}
}
